src/opm/output/eclipse/EclipseIO.cpp

```diff
@@ -123,35 +123,34 @@ class RFT {
 {}
 
 
-void RFT::writeTimeStep( std::vector< const Well* > wells,
+void RFT::writeTimeStep( const Schedule& schedule,
                          const EclipseGrid& grid,
-                         int report_step,
+                         std::size_t report_step,
                          time_t current_time,
                          double days,
                          const UnitSystem& units,
                          data::Wells wellDatas) {
     using rft = ERT::ert_unique_ptr< ecl_rft_node_type, ecl_rft_node_free >;
+    const auto& rft_config = schedule.rftConfig();
+    auto well_names = schedule.wellNames(report_step);
+    if (!rft_config.active(report_step))
+        return;
 
     fortio_type * fortio;
-    int first_report_step = report_step;
-
-    for (const auto* well : wells)
-        first_report_step = std::min( first_report_step, well->firstRFTOutput());
 
-    if (report_step > first_report_step)
+    if (report_step > rft_config.firstRFTOutput())
         fortio = fortio_open_append( filename.c_str() , fmt_file , ECL_ENDIAN_FLIP );
     else
         fortio = fortio_open_writer( filename.c_str() , fmt_file , ECL_ENDIAN_FLIP );
 
-    for ( const auto& well : wells ) {
-        if( !( well->getRFTActive( report_step )
-            || well->getPLTActive( report_step ) ) )
-            continue;
+    for ( const auto& well_name : well_names ) {
 
-        auto* rft_node = ecl_rft_node_alloc_new( well->name().c_str(), "RFT",
-                current_time, days );
+        if (!(rft_config.rft(well_name, report_step) || rft_config.plt(well_name, report_step)))
+            continue;
 
-        const auto& wellData = wellDatas.at(well->name());
+        auto* well = schedule.getWell(well_name);
+        auto* rft_node = ecl_rft_node_alloc_new( well_name.c_str(), "RFT", current_time, days );
```

Comment:
Would it be possible put the `rft_node` (i.e., the `alloc_new()` call) directly into the `rft` smart pointer type here?  Doing so would prevent a memory leak if `wellData.empty()` and simplify cleanup in the case of an exception.